src/passes/wasm-intrinsics.wat

```diff
@@ -142,6 +143,41 @@
    (local.get $var$1)
   )
  )
+ (func $__wasm_i64_mulh (; 5 ;) (type $0) (param $var$0 i64) (param $var$1 i64) (result i64)
```

Comment:
was this generated using Rust like the comment on lines 1-6 says? Or some other way?

Reply:
It's generated from AssemblyScript file:
```ts
function umulh64(u: u64, v: u64): u64 {
  var u0 = u & 0xFFFFFFFF;
  var v0 = v & 0xFFFFFFFF;

  var u1 = u >> 32;
  var v1 = v >> 32;

  var l = u0 * v0;
  var t = u1 * v0 + (l >> 32);
  var w = u0 * v1 + (t & 0xFFFFFFFF);

  t >>= 32;
  w >>= 32;

  return u1 * v1 + t + w;
}
```
Rust generate more ore less the same codegen. But I could fork original gist and add Rust's solution

Reply:
I think it's ok as it is. Just please update the lines of documentation there, and link to this code sample there (can link to this PR).

Reply:
Done!